src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -235,6 +235,22 @@ public sealed class StructType : DataType
         /// </summary>
         public List<StructField> Fields { get; private set; }
 
+        /// <summary>
+        /// Constructor for StructType class.
+        /// </summary>
+        JvmObjectReference IJvmObjectReferenceProvider.Reference
```

Comment:
One reason I am not sure if this is the right approach or not is that we have `public StructType(IEnumerable<StructField> fields)`. For example, if I create a `StructType` object, I can have two objects one in JVM and on in CLR. 

If you look at other class that derives `IJvmObjectReferenceProvider`, it's just a thin wrapper.

@suhsteve did we compare with PySpark's approach?

Reply:
Yes I've pointed out two approaches that can possibly be taken.  The Jvm object route, or checking the pyspark implementation.

Reply:
JVM object route doesn't seem to make sense, right?

Reply:
Yes I don't think we need a JVM object for every structtype/structfield.  If we went that route, that would also mean that when we deserialize rows, we would be creating JVM StructType object(s ) for every batch.

Reply:
Yes, when I was checking PySpark's approach, I did encounter some issues with `SparkSession` and `parseDataType`, then I tried Jvm object route. I will explore more of PySpark's approach. Thanks!

Reply:
I think you should explore the [parseDatatType](https://github.com/apache/spark/blob/fff2e847c2fb58e4ef50063e70fa0053498a4109/python/pyspark/sql/readwriter.py#L94) route again.